### PR TITLE
Include player nicknames in dialogue

### PR DESCRIPTION
**Pull request by menynexi**
add functions deconstructsSentence(), reconstructsSentence(words) to display players nicknames in game while still using AddStaticData.html to query in data

### CODE REVIEW

File: src/main/webapp/completedPath.html [hunk 1]
```diff
@@ -21,11 +21,10 @@ <h2 class="step-outline">THE END?</h2>
 
   <div class="step-wrap, step-dialogue">
       <div>
-          <!--TODO:add graduation speech as dialogue-->
-          <p id="dialogue-container"></p>
+          <p id="dialogue-containerX"></p>
```
Comment (lzoebaker) [inline src/main/webapp/completedPath.html:24]: [something]X is not really a descriptive/ meaningful naming scheme.

File: src/main/webapp/gameHelper.js [hunk 2]
```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
...
+/*
+ * This function reconstructs words to make a sentence
+ */
+function reconstructsSentence(words) {
+  var i;
+  var sentence = '';
+  for(i = 0; i < words.length;i++) {
+    sentence = sentence + words[i] + ' ';
+  }
+  dialogueArrayX[dialogueRegexX] = sentence;
```
Comment (lzoebaker) [inline src/main/webapp/gameHelper.js:214]: Replace "X" with something more meaningful.

File: src/main/webapp/completedPath.html (hunk 1 above)
Comment (lattai) [reply src/main/webapp/completedPath.html:24]: +1

File: src/main/webapp/completedPath.js [hunk 3]
```diff
@@ -14,4 +14,5 @@
 function loadFunctions() {
   playmusic();
   getImage();
-}
+  getDialogue();
+}
```
Comment (lattai) [inline src/main/webapp/completedPath.js:18]: New Line

File: src/main/webapp/gameHelper.js [hunk 4]
```diff
@@ -34,18 +38,20 @@ function handleResponse(response) {
...
+  deconstructsSentence();
   quoteContainer.innerText = dialogueArray[dialogueRegex];
 }
- 
+
 function nextLine() {
   const quoteContainer = document.getElementById('dialogue-container');
   if (dialogueRegex < dialogueArray.length-1){
     dialogueRegex ++;
+    deconstructsSentence();
```
Comment (lattai) [inline src/main/webapp/gameHelper.js:54]: Really picky, but I don't know if deconstruct is the best verb? I think splitSentence is more accurate

File: src/main/webapp/gameHelper.js [hunk 5]
```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
   fetch('/promotion-threshold', {method: 'POST', body: params});
   window.location='promotionquiz.html';
 }
+
+/*
+ * This function deconstructs the current dialogue displayed and searches for PLAYER_NAME_PARAMETER
+ * Then replaces it with playerNickname
+ */
+function deconstructsSentence() {
```
Comment (lattai) [inline src/main/webapp/gameHelper.js:193]: There is a string method called replace/ replaceAll that might do just what you're doing here in two methods in one line.

File: src/main/webapp/gameHelper.js [hunk 6]
```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
...
+ * This function reconstructs words to make a sentence
+ */
+function reconstructsSentence(words) {
+  var i;
+  var sentence = '';
+  for(i = 0; i < words.length;i++) {
+    sentence = sentence + words[i] + ' ';
+  }
+  dialogueArrayX[dialogueRegexX] = sentence;
+}
```
Comment (lattai) [inline src/main/webapp/gameHelper.js:215]: New Line eof

File: src/main/webapp/gameHelper.js [hunk 7]
```diff
@@ -16,6 +16,10 @@ const FINAL_STAGE_REDIRECTION = 'completedPath.html';
 const MUSIC_ICON= 'musicIcon';
 const UNMUTE_ICON = '<img src="icons/unmute.gif" alt="unmuted icon">';
 const MUTE_ICON = '<img src="icons/mute.png" alt="muted icon">';
+var PLAYER_NAME_PARAMATER = 'playerNameHere';
+var words
+//TODO: fetch player name from a servlet and replace playerNickname
```
Comment (WillHipschman) [inline src/main/webapp/gameHelper.js:21]: Link to github issue

File: src/main/webapp/gameHelper.js [hunk 8]
```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
   fetch('/promotion-threshold', {method: 'POST', body: params});
   window.location='promotionquiz.html';
 }
+
+/*
+ * This function splits sentences into words the current dialogue displayed and searches for PLAYER_NAME_PARAMETER
```
Comment (WillHipschman) [inline src/main/webapp/gameHelper.js:190]: This comment doesn't make sense (missing words? extra words?)

"splits the sentences into words the current dialogue displayed" <-- I don't know what this means

File: src/main/webapp/gameHelper.js [hunk 9]
```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
...
+ * Then replaces it with playerNickname
+ */
+function splitSentenceToWords() {
+  words = dialogueArray[dialogueRegex].split(' ');
+  var wordIterator;
+  for(wordIterator = 0; wordIterator < words.length; wordIterator++) {
+    if(words[wordIterator] == PLAYER_NAME_PARAMATER) {
+      //TODO: replace playerNickname with fetched player.getname();
```
Comment (WillHipschman) [inline src/main/webapp/gameHelper.js:198]: replace TODO w/ github issue

**Pull request merged**